pkg/core/aliaslist.go

```diff
@@ -0,0 +1,133 @@
+package core
+
+import (
+	"sync"
+
+	"github.com/rs/zerolog/log"
+	"github.com/zinclabs/zinc/pkg/metadata"
+	"github.com/zinclabs/zinc/pkg/zutils"
+)
+
+var ZINC_INDEX_ALIAS_LIST AliasList
+
+type AliasList struct {
+	lock    sync.RWMutex
+	Aliases map[string][]string
+}
+
+func NewAliasList() *AliasList {
+	return &AliasList{Aliases: map[string][]string{}}
+}
+
+func (al *AliasList) AddIndexesToAlias(alias string, indexes []string) error {
+	al.lock.Lock()
+	al.Aliases[alias] = append(al.Aliases[alias], indexes...)
+
+	err := metadata.Alias.Set(al.Aliases)
+	if err != nil {
+		log.Err(err).Msg("failed to save alias in metadata after add operation")
+		al.lock.Unlock()
+		return err
+	}
+
+	al.lock.Unlock()
+	return nil
+}
+
+func (al *AliasList) RemoveIndexesFromAlias(alias string, removeIndexes []string) error {
+	al.lock.Lock()
+
+	indexes, ok := al.Aliases[alias]
+	if !ok {
+		al.lock.Unlock()
+		return nil
+	}
+
+outer:
+	for _, removeIndex := range removeIndexes {
+		for i, s := range indexes {
+			if s == removeIndex {
+				indexes = append(indexes[:i], indexes[i+1:]...)
+				continue outer
```

Comment:
This section have bug, if `i` is the last item, `i+1` will panic.

Reply:
Got it

Reply:
Fixed